tern/analyze/docker/analyze.py

```diff
@@ -142,15 +142,19 @@ def analyze_subsequent_layers(image_obj, shell, master_list, redo, dfobj=None,
                 if dfile_lock:
                     # collect list of RUN commands that could install pkgs
                     run_dict = d_file.get_run_layers(dfobj)
-                    for package in image_obj.layers[curr_layer].packages:
-                        # check that package is in current dfobj RUN line
-                        if d_file.package_in_dockerfile(
-                                run_dict[curr_layer - 1], package.name):
-                            d_file.expand_package(
-                                run_dict[curr_layer - 1], package.name,
-                                package.version,
-                                command_lib.check_pinning_separator(
-                                    pkg_listing))
+                    # use the run_dict to get list of packages being installed
+                    install_list = d_file.get_install_packages(
+                        run_dict[curr_layer - 1])
+                    for install_pkg in install_list:
+                        for layer_pkg in image_obj.layers[curr_layer].packages:
+                            if install_pkg == layer_pkg.name:
```

Comment:
how about `if install_pkg in [p.name for p in image_obj.layers[curr_layer].packages]`?

Reply:
install_pkg is an element in install_list -- so it has no value to compare to the loop in this suggestion. Also, if we do this then we are basically just doing what we did before, looking through the packages in the layer to see if they are in the dockerfile RUN line. We want to only look at the packages that are being installed in the RUN line first and then check if they are in the layer.

Reply:
How do we get the `layer_pkg` version this way? using `p.version` in the `expand_function` is unrecognized.

Reply:
Agree your suggestion is more compact, but we need the `layer_pkg` property data available inside the if statement.

Reply:
OK, got it.